### ISSUE TITLE
Error  with function analyseWGCNA()

### ISSUE THREAD

**Issue by GERMAN00VP** [mortazavilab/PyWGCNA#48]
HI I AM TRYING TO FIND THE CORRELATION OF THE MODULES OF MY WGCNA OBJECT WITH THE TRAITS OF MY METADATA ,  BUT I GET THE FOLLOWING ERROR, DO ANYONE KNOW HOW TO SOLVE IT? THANK YOU.

 pyWGCNA_TCGA_PAAD.analyseWGCNA() 

Analysing WGCNA...
Calculating module trait relationship ...
	Done..

Adding (signed) eigengene-based connectivity (module membership) ...
	Done..

plotting module heatmap eigengene...
	Done..

plotting module barplot eigengene...
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[10], line 1
----> 1 pyWGCNA_TCGA_PAAD.analyseWGCNA()

File [~/miniconda3/envs/machine_learning/lib/python3.11/site-packages/PyWGCNA/wgcna.py:447](https://file+.vscode-resource.vscode-cdn.net/media/victor/USB_GERMAN/RNASEQ/WGCNA/~/miniconda3/envs/machine_learning/lib/python3.11/site-packages/PyWGCNA/wgcna.py:447), in WGCNA.analyseWGCNA(self, order, geneList, show)
    445     print(f"{OKCYAN}plotting module barplot eigengene...{ENDC}")
    446     for module in modules:
--> 447         self.barplotModuleEigenGene(module, metadata, colorBar=metadata[-1], show=True)
    448     print("\tDone..\n")
    450 if self.save:

File [~/miniconda3/envs/machine_learning/lib/python3.11/site-packages/PyWGCNA/wgcna.py:2946](https://file+.vscode-resource.vscode-cdn.net/media/victor/USB_GERMAN/RNASEQ/WGCNA/~/miniconda3/envs/machine_learning/lib/python3.11/site-packages/PyWGCNA/wgcna.py:2946), in WGCNA.barplotModuleEigenGene(self, moduleName, metadata, combine, colorBar, show)
   2944 df['all'] = df['all'].apply(lambda x: x[1:])
   2945 cat = pd.DataFrame(pd.unique(df['all']), columns=['all'])
-> 2946 cat[metadata] = cat['all'].str.split('_', expand=True)
   2947 ybar = df[['all', 'eigengeneExp']].groupby(['all']).mean()['eigengeneExp']
   2948 ebar = df[['all', 'eigengeneExp']].groupby(['all']).std()['eigengeneExp']

File [~/miniconda3/envs/machine_learning/lib/python3.11/site-packages/pandas/core/frame.py:3968](https://file+.vscode-resource.vscode-cdn.net/media/victor/USB_GERMAN/RNASEQ/WGCNA/~/miniconda3/envs/machine_learning/lib/python3.11/site-packages/pandas/core/frame.py:3968), in DataFrame.__setitem__(self, key, value)
   3966     self._setitem_frame(key, value)
   3967 elif isinstance(key, (Series, np.ndarray, list, Index)):
-> 3968     self._setitem_array(key, value)
   3969 elif isinstance(value, DataFrame):
   3970     self._set_item_frame_value(key, value)
...
    402 else:
    403     # Missing keys in columns are represented as -1
    404     if len(columns.get_indexer_non_unique(key)[0]) != len(value.columns):

ValueError: Columns must be same length as key

**Comment by nargesr** [comment]
Hi, 

did you add your metadata using updateSampleInfo()?
can you tell me what's in the pyWGCNA_TCGA_PAAD.datExpr.obs?

**Comment by GERMAN00VP** [comment]
Yes I did update the sample and gene info following all the steps of the quick start tutorial up to the one wich gave me the error. I will upload a photo of the pyWGCNA_TCGA_PAAD.datExpr.obs dataframe and the sample info file so that you can explore the metadata I am using. Thank you for your time and answers.



[sampleInfo.csv](https://github.com/mortazavilab/PyWGCNA/files/11832378/sampleInfo.csv)

![image](https://github.com/mortazavilab/PyWGCNA/assets/121015636/61e91b61-5b28-4100-915e-480394d0b2f2)

**Comment by nargesr** [comment]
oh, it's probably because you have a dash in your column name like "Cluster_0"
I will try to fix this problem but meanwhile, if you can change your column names to be dash-free, it will probably solve your problem for now

Thank you for catching that! I'll try to see how I can fix that :)
